### PR TITLE
Resolves Validation for Encrypted Numeric and Alpha Custom Fields

### PR DESCRIPTION
**Pull request by spencerrlongg**
# Description
This adds two validation rules: NumericEncrypted and AlphaEncrypted - the rules are written in the new object syntax. They do the same validation as the basic Laravel rules, but add a step to decrypt the value. The value is decrypted only in the context of the Rule class, and remains encrypted when saving to the db. 

https://github.com/user-attachments/assets/4b6d5808-d0ec-48f6-ac87-8f5fad7a410c

Fixes SC-27213

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: app/Rules/AlphaEncrypted.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        try {
+            $attributeName = trim(preg_replace('/_+|snipeit|\d+/', ' ', $attribute));
+            $decrypted = Crypt::decrypt($value);
+            if (!ctype_alpha($decrypted) && !is_null($decrypted)) {
+                $fail(trans('validation.custom.alpha_encrypted', ['attribute' => $attributeName]));
+            }
+        } catch (\Exception $e) {
+            report($e);
+            $fail('something went wrong.');
```
Comment (marcusmoore) [inline app/Rules/AlphaEncrypted.php]: Can we use a validation string for this? We probably have one we can use for this.

File: app/Rules/NumericEncrypted.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        try {
+            $attributeName = trim(preg_replace('/_+|snipeit|\d+/', ' ', $attribute));
+            $decrypted = Crypt::decrypt($value);
+            if (!is_numeric($decrypted) && !is_null($decrypted)) {
+                $fail(trans('validation.custom.numeric_encrypted', ['attribute' => $attributeName]));
+            }
+        } catch (\Exception $e) {
+            report($e->getMessage());
+            $fail('something went wrong');
```
Comment (marcusmoore) [inline app/Rules/NumericEncrypted.php]: Can we use a validation string for this? We probably have one we can use for this.

**Review by marcusmoore — CHANGES_REQUESTED** [review]
Tested and works 👍🏾 

We talked about it while pairing and you already know about the translation strings 😉

**Review by marcusmoore — APPROVED** [review]

File: resources/lang/en-US/validation.php [hunk 3]
```diff
@@ -187,6 +187,8 @@
 
     'custom' => [
         'alpha_space' => 'The :attribute field contains a character that is not allowed.',
+        'alpha_encrypted'   => 'The :attribute field should be alphabetic characters.',
```
Comment (snipe) [inline resources/lang/en-US/validation.php]: We should be able to use the existing validation strings for this, no?

Comment (spencerrlongg) [reply resources/lang/en-US/validation.php]: Derp, yeah, resolved.

**Pull request merged**